Search/bfsnew.cpp

```diff
@@ -0,0 +1,58 @@
+#include <iostream>
+using namespace std;
+int queue[5];
+int a[5];
+int root;
+int L;
+int R;
+int front=0;
+int end=0;
+void BFS();
+void push();
+void pop();
+
+int main()
+{
+	for(int i=0;i<5;i++)
+	{
+		cin>>a[i];
+	}
+	cout<<"\nPrinting in bfs order";
+	BFS();
+	
+	return 0;
+ } 
+ void BFS()
+ {
+
+ 	queue[0]=a[root];
```

Comment:
you have not initialized **root** so it may take any **random junk value** as its initial value. Assume **root** take 1583 as its initial value, then _a[**1583**]_ may crash the program because size of **a** is 5.

[learncpp.com - uninitialized variables & undefined behavior](https://www.learncpp.com/cpp-tutorial/uninitialized-variables-and-undefined-behavior/)